### PR TITLE
add ddcutil/1.2.1

### PR DESCRIPTION
**Pull request by daravi**
Specify library name and version:  **ddcutil/1.2.1**

Adding recipe for [ddcutil](https://github.com/rockowitz/ddcutil). (example usage of this library is to control display brightness on Linux)

---

- [✓] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [✓] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [✓] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [✓] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by daravi** [comment]
I don't know how to make a PR be blocked by another one, but this can only be merged after:
https://github.com/conan-io/conan-center-index/pull/8557

**Comment by daravi** [comment]
For some reason the build is looking for `libi2c.so.0`, while the `i2c-tools` package only has `libi2c.so` (and no symlinks) in the `lib` directory. Any ideas why/how this is happening?

**Comment by daravi** [comment]
@eirikb It seems that `i2c-tools` build produces:
```
libi2c.so -> libi2c.so.0.1.1
libi2c.so.0.1.1
```
But the consumer (here `ddcutil`) for some reason is expecting `libi2c.so.0`. So we need another symbolic link:
```
libi2c.so.0 -> libi2c.so.0.1.1
```
Is this is a bug in `i2c-tools` recipe? What would you recommend to resolve this issue?

Note: I am building on CentOS 8

**Comment by eirikb** [comment]
> @eirikb It seems that `i2c-tools` build produces:
> 
> ```
> libi2c.so -> libi2c.so.0.1.1
> libi2c.so.0.1.1
> ```
> 
> But the consumer (here `ddcutil`) for some reason is expecting `libi2c.so.0`. So we need another symbolic link:
> 
> ```
> libi2c.so.0 -> libi2c.so.0.1.1
> ```
> 
> Is this is a bug in `i2c-tools` recipe? What would you recommend to resolve this issue?
> 
> Note: I am building on CentOS 8

Must admit I haven't tested that much with shared. When we tested with ddcutil I believe we used static as well, that is the default.
Did a test now, and `-o i2c-tools:shared=True` does indeed produce libi2c.so.0.1.1, with libi2c.so as a symlink.  
The file that ends up in the package folder is libi2c.so, not a symlink, would expect ddcutil to pick up on this.

**Comment by SpaceIm** [comment]
@daravi try https://github.com/conan-io/conan-center-index/pull/8558

**Comment by daravi** [comment]
@SpaceIm I have tried that locally but as I mentioned on slack it is producing these files:
```
libi2c.so -> libi2c.so.0.1.1
libi2c.so.0.1.1
```
Whereas this build for some reason is looking for `libi2c.so.0`. (and my system package manage installs these two:
```
libi2c.so -> libi2c.so.0.1.1
libi2c.so.0.1.1
```

**Comment by SpaceIm** [comment]
Yes, it's fixed in the PR now.

File: recipes/ddcutil/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import os
+
+required_conan_version = ">=1.33.0"
+
+class DdcutilConan(ConanFile):
+    name = "ddcutil"
+    description = "ddcutil is a Linux program for managing monitor settings, such as brightness, color levels, and input source"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.ddcutil.com/"
+    topics = ("ddcutil", "display", "monitor", "settings", "Linux")
```
Comment (SSE4) [inline recipes/ddcutil/all/conanfile.py:12]:
```suggestion
    topics = ("ddcutil", "display", "monitor", "settings", "linux")
```

File: recipes/ddcutil/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            if tools.os_info.with_yum:
+                packages = ["hwdata", "libgudev"]
+            elif tools.os_info.with_apt:
+                packages = ["hwdata", "libgudev-1.0-dev"]
+            elif tools.os_info.with_pacman:
+                packages = ["hwids", "libgudev"]
+            elif tools.os_info.with_zypper:
+                packages = ["hwdata", "libgudev-1_0-0"]
+            else:
+                self.output.warn("Don't know how to install %s for your distro." % self.name)
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/ddcutil/all/conanfile.py:73]: libudev is our old friend on Linux, but strongly connected to systemd project. Gnome project separated it to a new project: https://github.com/GNOME/libgudev/blob/master/meson.build For our luck, we can build it, but it uses meson.
My suggest creating a separated package, not a system package.

**Comment by eirikb** [comment]
Hi. Any news on this?

**Comment by uilianries** [comment]
@eirikb we need some champion who is interested to decouple udev and hwdata from this recipe. The SystemPackageTool is not allowed in regular recipes, need to be moved to a system recipe instead.

**Pull request closed without merging**